blockchain/message_pool/src/msgpool.rs

```diff
@@ -70,6 +75,45 @@ impl MsgSet {
         self.msgs.insert(m.sequence(), m);
         Ok(())
     }
+    pub fn rm(&mut self, nonce: u64, applied: bool) {
+        let m = if let Some(m) = self.msgs.get(&nonce) {
+            m
+        } else {
+            if applied && nonce >= self.next_sequence {
+                self.next_sequence = nonce + 1;
+                while self.msgs.get(&self.next_sequence).is_some() {
+                    self.next_sequence += 1;
+                }
+            }
+            return;
+        };
+        self.required_funds -= m.required_funds();
+        // guaranteed to not panic
+        self.msgs.remove(&nonce).unwrap();
```

Comment:
why not have the remove done above? The remove moves the removed value as a return (avoid a potential panic if logic changes also)